src/ffi.rs

```diff
@@ -19,40 +19,20 @@ impl NewtonPoint {
 }
 
 #[no_mangle]
-pub extern "C" fn newton_new_field(
-    g: f32,
-    solar_mass: f32,
-    min_dist: f32,
-    max_dist: f32,
-) -> *mut Field {
-    let field = Field::new(g, solar_mass, min_dist, max_dist);
-    let boxed = Box::new(field);
+pub extern "C" fn newton_new_environment() -> *mut Environment {
+    let environment = Environment::new();
+    let boxed = Box::new(environment);
     Box::into_raw(boxed)
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn newton_destroy_field(field: *mut Field) {
```

Comment:
Could you maybe explain `unsafe` ?

Reply:
You need to mark a function `unsafe` when it contains raw pointers, since these will not be automatically freed when the function returns. It is _unsafe_ because the user is now responsible for managing allocated memory (like in C), whereas this is normally handled by Rust.

Reply:
when is a pointer raw? because of `*mut` ?